ci-cd/Jenkinsfile.groovy

```diff
@@ -59,130 +52,249 @@ spec:
   volumes:
   - name: docker-graph-storage
     emptyDir: {}
-"""
+'''
+
+pipeline {
+    agent {
+        kubernetes {
+            yaml podSpec
         }
     }
-    stages {
 
-        // get the commit and version number for current release
-        stage('Prepare') {
-            steps {
-                script {
-                    commit = sh(returnStdout: true, script: 'git describe --always').trim()
-                }
-                script {
-                    version = sh(returnStdout: true, script: "cat pom.xml | grep \"<version>.*</version>\" | head -1 |awk -F'[><]' '{print \$3}'").trim()
-                }
-            }
-        }
+    environment {
+        dockerHub = 'overture/maestro'
+        gitHubRegistry = 'ghcr.io'
+        gitHubRepo = 'overture-stack/maestro'
+        githubPackages = "${gitHubRegistry}/${gitHubRepo}"
+
+        commit = sh(
+            returnStdout: true,
+            script: 'git describe --always'
+        ).trim()
+
+        version = readMavenPom().getVersion()
+    }
 
+    options {
+        timeout(time: 30, unit: 'MINUTES')
+        timestamps()
+    }
+
+    stages {
         // run tests and package
         stage('Test') {
             steps {
                 container('jdk') {
-                    sh "./mvnw test package"
+                    sh './mvnw test package'
                 }
             }
         }
 
         // run tests and package
-        stage('build') {
+        stage('build image') {
             steps {
                 container('docker') {
-                    // the network=host needed to download dependencies using the host network (since we are inside 'docker'
-                    // container)
-                    sh "docker build --network=host -f ci-cd/Dockerfile . -t overture/maestro:${version}-${commit} -t ${dockerRepo}:${version}-${commit}"
+                    // the network=host needed to download dependencies using the host network
+                    // (since we are inside the 'docker' container)
+                    sh "docker build --network=host -f ci-cd/Dockerfile . -t maestro:${commit}"
+                }
+            }
+        }
+
+        stage('Publish images') {
+            when {
+                anyOf {
+                    branch 'develop'
+                    branch 'main'
+                    branch 'master'
+                    branch 'test'
```

Comment:
why `test` branch?

Reply:
Aside from `test` being my usually preferred branch name for this sort of devops chore, that's an attempt at a "declarative" example of how to trigger these from a custom branch.
e.g. add your branchname in a new line wherever you see `branch 'test'` and related.

Consistent with our conversations over this approach with @lrivera-oicr and @UmmulkiramR, this serves as a reminder/guide for the procedure to build testing snapshots for us all.